### PR TITLE
Updates compiler error E0040 with new format

### PR DESCRIPTION
**Pull request by shri3k**
Addresses #35208 as part of #35233.
r? @GuillaumeGomez

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
Thanks!

Can you update the corresponding test file in compile-fail directory please?

**Comment by shri3k** [comment]
Yeah, it seems I forgot to update on both my PRs. I'll go ahead and update this later tonight. Thank you for catching it. 😳

**Comment by shri3k** [comment]
I was able to find only one test file corresponding to this error. Updated it.

r? @GuillaumeGomez

**Comment by GuillaumeGomez** [comment]
Thanks!

@bors: r+ rollup

File: src/test/compile-fail/E0040.rs [hunk 1]
```diff
@@ -20,5 +20,7 @@ impl Drop for Foo {
 
 fn main() {
     let mut x = Foo { x: -7 };
-    x.drop(); //~ ERROR E0040
+    x.drop();
```
Comment (KiChjang) [inline src/test/compile-fail/E0040.rs]: Trailing whitespace on this line.

Comment (shri3k) [inline src/test/compile-fail/E0040.rs]: Updated. :sweat_smile:  Sorry.

**Pull request merged**